### PR TITLE
deepin.gio-qt: enable build docs

### PR DESCRIPTION
**Pull request by wineee**
###### Description of changes
https://github.com/linuxdeepin/dde-nixos

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/deepin/library/gio-qt/default.nix [hunk 1]
```diff
@@ -20,15 +22,23 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-dlY1CTlXywgGZUonBBe3cDwx8h2xXrPY6Ft/D59nlug=";
   };
 
-  nativeBuildInputs = [ cmake pkg-config wrapQtAppsHook ];
+  patchPhase = ''
```
Comment (Artturin) [inline pkgs/desktops/deepin/library/gio-qt/default.nix:25]:
```suggestion
  postPatch = ''
```

phases shouldn't be overriden

File: pkgs/desktops/deepin/library/gio-qt/default.nix [hunk 2]
```diff
@@ -20,15 +22,23 @@ stdenv.mkDerivation rec {
...
+    # qt.qpa.plugin: Could not find the Qt platform plugin "minimal"
+    # A workaround is to set QT_PLUGIN_PATH explicitly
+    export QT_PLUGIN_PATH=${qtbase.bin}/${qtbase.qtPluginPrefix}
```
Comment (SuperSandro2000) [inline pkgs/desktops/deepin/library/gio-qt/default.nix:28]: Ideally we set this in the pre phase where it is required.

File: pkgs/desktops/deepin/library/gio-qt/default.nix [hunk 3]
```diff
@@ -20,15 +22,23 @@ stdenv.mkDerivation rec {
...
+    substituteInPlace CMakeLists.txt --replace "qhelpgenerator" "${qttools.dev}/bin/qhelpgenerator"
```
Comment (SuperSandro2000) [inline pkgs/desktops/deepin/library/gio-qt/default.nix:29]: Isn't this taken from PATH?

File: pkgs/desktops/deepin/library/gio-qt/default.nix (hunk 2 above)
Comment (wineee) [reply pkgs/desktops/deepin/library/gio-qt/default.nix]: Should I use preConfigure ?

File: pkgs/desktops/deepin/library/gio-qt/default.nix (hunk 3 above)
Comment (wineee) [reply pkgs/desktops/deepin/library/gio-qt/default.nix]: Should add qttools.dev instead of qttools in `nativeBuildInputs`,  updated already

**Pull request merged**